common/src/main/java/org/apache/nemo/common/DirectByteBufferInputStream.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.nemo.common;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.List;
+
+/**
+ * This class is a customized input stream implementation which reads data from
+ * list of {@link ByteBuffer}.
+ */
+public class DirectByteBufferInputStream extends InputStream {
+  private List<ByteBuffer> bufList;
```

Comment:
I think you're right, the name of the class implies that this is only for `DirectByteBuffer`. I guess it is better to change the name of the class to `ByteBufferInputStream` because `DirectByteBuffer` class is not public and it can only be constructed by calling `allocateDirect()` method in `ByteBuffer`. It is possible to check whether the `ByteBuffer` is direct or not, but I am afraid it has no much meaning in checking it when it is already written(in on-heap or off-heap) and we are intending to *read* it. I will change the name to make it more straightforward and be used to read the on-heap `ByteBuffer` in future implementation. Thanks!

Reply:
Agreed.
I am also fine with keeping the current class name `DirectByteBufferInputStream`, as we also have `DirectByteBufferOutputStream`.